### PR TITLE
Diffusers/thuchk

### PR DESCRIPTION
**Pull request by thuc-moreh**
- Add memory record file. 
- Add more available models for each task
- Fix the error when running single model
- Fix Runtimeerror Permission denied for textual inversion task

### CODE REVIEW

**Review by kimmorehvn — APPROVED** [review]

File: examples/controlnet/train.sh [hunk 1]
```diff
@@ -13,14 +32,12 @@ else
...
 
-moreh-switch-model -M 1 
-
 # Run training script
 echo "# ========================================================= #"
 echo "training ${model_name}.."
 conda run -n ${env_name} python3 train_controlnet.py \
-    --pretrained_model_name_or_path ${model_name}  \
-    --controlnet_model_name_or_path lllyasviel/sd-controlnet-hed \
+    --pretrained_model_name_or_path runwayml/stable-diffusion-v1-5  \
```
Comment (yeomni) [inline examples/controlnet/train.sh:36]: Is this one the only option for pretrained model? Wondering why this specific model name is hardcoded here.

File: examples/textual_inversion/textual_inversion.py [hunk 2]
```diff
@@ -930,7 +940,15 @@ def main():
                 commit_message="End of training",
                 ignore_patterns=["step_*", "epoch_*"],
             )
-
+    elapsed_time = time.time() - start_time
+    throughput = (len(train_dataset)*args.num_train_epochs) / elapsed_time
```
Comment (yeomni) [inline examples/textual_inversion/textual_inversion.py:944]: Should be much better if it shows throughput more frequently using something like log_interval option during the training process unless such more frequent throughput calculation makes the whole process too slow.

File: examples/controlnet/train.sh [hunk 3]
```diff
@@ -1,10 +1,29 @@
...
+
+if [[ "$@" == *"--batch_size"* ]]; then 
+  batch_size="$(echo "$@" | sed -n 's/.*--batch_size \([^ ]*\).*/\1/p')"
+fi
+
+output_dir=outputs
+if [[ "$@" == *"--output_dir"* ]]; then 
+  output_dir="$(echo "$@" | sed -n 's/.*--output_dir \([^ ]*\).*/\1/p')"
+fi
+
```
Comment (loctxmoreh) [inline examples/controlnet/train.sh]: This method of parsing arguments looks pretty cumbersome. Have you considered using short args (like `-m`, `-t`, `-b`) and parsing them using `getopts` like [this](https://github.com/moreh-dev/transformers/blob/8428265f1879d34b56e2f24387ee898f2b79da0a/examples/pytorch/language-modeling/train_clm.sh#L1)?

File: examples/controlnet/train.sh (hunk 1 above)
Comment (thuc-moreh) [reply examples/controlnet/train.sh:36]: Controlnet task uses two types of models: stable diffusion and contronet at the same time. The stable diffusion models are trained specifically for the text_to_image task. Therefore, in the controlnet task, the script is modified to accommodate the different controlnet models.

File: examples/textual_inversion/textual_inversion.py (hunk 2 above)
Comment (thuc-moreh) [reply examples/textual_inversion/textual_inversion.py:944]: I will implement this in next PR

File: examples/controlnet/train.sh (hunk 3 above)
Comment (thuc-moreh) [reply examples/controlnet/train.sh]: Thanks. I tested out the method you recommended, and it worked well. The script looks much neater now

**Pull request merged**